src/components/common/Actions.vue

```diff
@@ -41,28 +65,77 @@
 </template>
 
 <script lang="ts">
-import { Component, Vue, Watch } from 'vue-property-decorator'
+// Libraries
+import { Component, Vue } from 'vue-property-decorator'
+import { State, Getter, Action } from 'vuex-class'
+
+// Interfaces
+import { StateModelIF, GetterIF, ActionBindingIF } from '@/interfaces'
 
 @Component
 export default class Actions extends Vue {
-  private get showBackBtn (): boolean {
-    return (this.$route.meta.step > 1)
+  // Global state
+  @State stateModel!: StateModelIF
+
+  // Global getters
+  @Getter isEntityType!: GetterIF
+  @Getter isShowBackBtn!: GetterIF
+  @Getter isShowReviewConfirmBtn!: GetterIF
+  @Getter isShowFilePayBtn!: GetterIF
+  @Getter isEnableFilePayBtn!: GetterIF
+  @Getter isBusySaving!: GetterIF
+
+  // Global actions
+  @Action setIsSaving!: ActionBindingIF
+  @Action setIsSavingResuming!: ActionBindingIF
+  @Action setIsFilingPaying!: ActionBindingIF
+
+  /**
+   * Method called when Cancel button is clicked.
+   */
+  private onCancel (): void {
+    const authUrl = sessionStorage.getItem('AUTH_URL')
+    // assume Auth URL is always reachable
+    window.location.assign(authUrl)
+  }
+
+  /**
+   * Method called when Save button is clicked.
+   * @returns A promise (ie, this is an async method).
+   */
+  private async onClickSave (): Promise<void> {
+    this.setIsSaving(true)
+    await this.sleep(1000)
+    this.setIsSaving(false)
   }
 
-  private get showReviewConfirmBtn (): boolean {
-    return (this.$route.meta.step < 2)
+  /**
+   * Method called when Save and Resume button is clicked.
+   * @returns A promise (ie, this is an async method).
+   */
+  private async onClickSaveResume (): Promise<void> {
+    this.setIsSavingResuming(true)
+    await this.sleep(1000)
+    this.setIsSavingResuming(false)
```

Comment:
This is neat! but why not timeout the second method call to `false`? Do we expect to be using the sleep() method in more places than this?

Reply:
This was just a quick hack to simulate something happening.

Yes, the second method could have been called after a timeout, but I think this is easier to understand (and will probably be the same code when we're done: simply replace the sleep by the actual work).

Reply:
sounds great! 👍